plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtDomainXMLParser.java

```diff
@@ -234,6 +256,10 @@ public Integer getVncPort() {
         return diskDefs;
     }
 
+    public List<ChannelDef> getChannels() {
+        return channels;
```

Comment:
Please make a defensive copy of the list before returning to avoid side-effects on this class by potential mutations of the list by the caller.  Ideally, return it as a ``Collections.unmodifiableList(channels)``.